app/helpers/categories_helper.rb

```diff
@@ -0,0 +1,21 @@
+module CategoriesHelper
+  def category_name
+    if @subcategory && @category
+      @subcategory.name
+    else
+      @category.name
+    end
+  end
+
+  def category_breadcrumb
+    output = ""
+    output += link_to "Home", root_path
+    output += "<span>&nbsp;»&nbsp;</span>"
+    output += link_to @category.name, category_path(@category.id)
```

Comment:
**Nitpick - Code Quality**
I understand you are trying to return a dynamic html string in `category_breadcrumb`. This can be conveniently done in a partial rather than in a module